src/utils/loadable.ts

```diff
@@ -28,7 +36,17 @@ export function loadable<Value>(anAtom: Atom<Value>): Atom<Loadable<Value>> {
           return LOADING_LOADABLE
         }
 
-        const cachedErrorLoadable = errorLoadableCache.get(error as Error)
+        let errorObject: object
+        if (typeof error === 'object') {
+          errorObject = error as object
+        } else if (ref.err === error) {
+          errorObject = ref.obj
+        } else {
+          errorObject = new Error(error as string)
```

Comment:
This seems a bit unintuitive that everywhere else Jotai keeps primitive values in tact, but here it converts them into error objects.

This also won't get cached because it's making a new error on every call.

Maybe this should be replaced with a refAtom after all? Although after the fix in #705, the caching may not be strictly necessary anymore?

Reply:
> Maybe this should be replaced with a refAtom after all?

This is true. I just wanted to keep the nice WeakMap approach, but didn't make sense...
Okay, let me refactor it again. btw, `refAtom` is a real hack and may not work in concurrent rendering. We need to revisit it. There are about three functions using it.

> Although after the fix in #705, the caching may not be strictly necessary anymore?

I don't think this has something to do with #705.